AppleALC/ALCUserClient/ALCUserClient.cpp

```diff
@@ -0,0 +1,86 @@
+//
+//  ALCUserClient.cpp
+//  AppleALC
+//
+//  Created by Nick on 10/14/20.
+//  Copyright © 2020 vit9696. All rights reserved.
+//
+
+#include <IOKit/IOLib.h>
+
+#include "ALCUserClient.hpp"
+
+OSDefineMetaClassAndStructors(ALCUserClient, IOUserClient);
+
+const IOExternalMethodDispatch ALCUserClient::sMethods[kNumberOfMethods] = {
+	{ //kMethodExecuteVerb
+		(IOExternalMethodAction) &ALCUserClient::methodExecuteVerb,			// Method pointer
+		3,                                                          		// Num of scalar input values
+		0,                                                          		// Num of struct input values
+		0,                                                          		// Num of scalar output values
+		0                                                           		// Num of struct output values
+	}
+};
+
+IOReturn ALCUserClient::externalMethod(uint32_t selector, IOExternalMethodArguments* arguments, IOExternalMethodDispatch* dispatch, OSObject* target, void* reference) {
+	if (selector >= kNumberOfMethods)
+		return kIOReturnUnsupported;
+	
+	dispatch = (IOExternalMethodDispatch*) &sMethods[selector];
+	
+	if (selector == kMethodExecuteVerb) {
```

Comment:
Maybe drop the condition and always assign mTarget? Easier to handle.

Reply:
Yeah this would be better as we are never sending anything else as a parameter from user-space client.